software/kernel/signpost_drivers/src/xdot.rs

```diff
@@ -0,0 +1,327 @@
+        }
+    }
+}
+
+impl<'a, U: UARTAdvanced> Driver for Console<'a, U> {
+    fn allow(&self, appid: AppId, allow_num: usize, slice: AppSlice<Shared, u8>) -> ReturnCode {
+        match allow_num {
+            // Allow a read buffer
+            0 => {
+                self.apps
+                    .enter(appid, |app, _| {
+                        app.read_buffer = Some(slice);
+                        app.read_idx = 0;
+                        ReturnCode::SUCCESS
+                    })
+                    .unwrap_or_else(|err| match err {
+                        Error::OutOfMemory => ReturnCode::ENOMEM,
+                        Error::AddressOutOfBounds => ReturnCode::EINVAL,
+                        Error::NoSuchApp => ReturnCode::EINVAL,
+                    })
+            }
+            // Allow a write buffer
+            1 => {
+                self.apps
+                    .enter(appid, |app, _| {
+                        app.write_buffer = Some(slice);
+                        ReturnCode::SUCCESS
+                    })
+                    .unwrap_or_else(|err| match err {
+                        Error::OutOfMemory => ReturnCode::ENOMEM,
+                        Error::AddressOutOfBounds => ReturnCode::EINVAL,
+                        Error::NoSuchApp => ReturnCode::EINVAL,
+                    })
+            }
+            _ => ReturnCode::ENOSUPPORT,
+        }
+    }
+
+    fn subscribe(&self, subscribe_num: usize, callback: Callback) -> ReturnCode {
+        match subscribe_num {
+            0 /* read line */ => {
+                // read line is not implemented for console at this time
+                ReturnCode::ENOSUPPORT
+            },
+            1 /* putstr/write_done */ => {
+                self.apps.enter(callback.app_id(), |app, _| {
+                    self.send_new(callback.app_id(), app, callback)
+                }).unwrap_or_else(|err| {
+                    match err {
+                        Error::OutOfMemory => ReturnCode::ENOMEM,
+                        Error::AddressOutOfBounds => ReturnCode::EINVAL,
+                        Error::NoSuchApp => ReturnCode::EINVAL,
+                    }
+                })
+            },
+            2 /* read automatic */ => {
+                self.apps.enter(callback.app_id(), |app, _| {
+                    app.read_callback = Some(callback);
+
+                    // only both receiving if we've got somewhere to put it
+                    app.read_buffer = app.read_buffer.take().map(|app_buf| {
+                        self.rx_buffer.take().map(|buffer| {
+                            /*
+                            let mut receive_len = app_buf.len();
+                            if receive_len > buffer.len() {
+                                receive_len = buffer.len();
+                            }*/
+                            //XXX: this could receive more than the app buffer...
+                            self.uart.receive_automatic(buffer, 10);
```

Comment:
This and the `receive_automatic` call in sara_u260 have different timeouts. You should put a comment to note the difference and explain why it changed from the console driver (in whichever changed).